### ISSUE TITLE
Xcode 8.1 requires project updates on Pods project using Cocoapods 1.1.1

### ISSUE THREAD

**Issue by davidbjames** [CocoaPods/CocoaPods#6117]
[Screenshot1](https://cl.ly/hxxj)
[Screenshot2](https://cl.ly/hxn7)

# Report

## What did you do?

ℹ Ran `pod update`

## What did you expect to happen?

ℹ Not to cause Xcode project settings to become invalid.

## What happened instead?

ℹ Project settings became invalid. (This is only a warning in Xcode and does not prevent me from doing my work.)

## CocoaPods Environment

ℹ ### Stack

```
   CocoaPods : 1.1.1
        Ruby : ruby 2.0.0p648 (2015-12-16 revision 53162) [universal.x86_64-darwin16]
    RubyGems : 2.0.14.1
        Host : Mac OS X 10.12.1 (16B2555)
       Xcode : 8.1 (8B62)
         Git : git version 2.9.3 (Apple Git-75)
Ruby lib dir : /System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib
Repositories : davidbjames - https://github.com/davidbjames/CocoaPods-Specs.git @ 6039ab2706c79cc33aad70dd15ed52c877658746
               master - https://github.com/CocoaPods/Specs.git @ 017bf33caf1cac28928d6537a4073c473468d091
```

### Installation Source

```
Executable Path: /usr/local/bin/pod
```

### Plugins

```
cocoapods-deintegrate : 1.0.1
cocoapods-plugins     : 1.0.0
cocoapods-search      : 1.0.0
cocoapods-stats       : 1.0.0
cocoapods-trunk       : 1.1.1
cocoapods-try         : 1.1.0
```

cc @DanToml

**Comment by davidbjames** [comment]
Additional note: Received this warning on `pod update`. This appears (though I'm not sure) to conflict with Xcode recommended settings. 

``` [!] The `Expand [Debug]` target overrides the `ALWAYS_EMBED_SWIFT_STANDARD_LIBRARIES` build setting defined in `Pods/Target Support Files/Pods-Expand/Pods-Expand.debug.xcconfig'. This can lead to problems with the CocoaPods installation
    - Use the `$(inherited)` flag, or
    - Remove the build settings from the target.

[!] The `Expand [Release]` target overrides the `ALWAYS_EMBED_SWIFT_STANDARD_LIBRARIES` build setting defined in `Pods/Target Support Files/Pods-Expand/Pods-Expand.release.xcconfig'. This can lead to problems with the CocoaPods installation
    - Use the `$(inherited)` flag, or
    - Remove the build settings from the target.```

**Comment by AnthonyMDev** [comment]
Yeah, this doesn't stop me from working, but is slightly annoying. I get a bunch of warnings, and then I update to recommended settings in Xcode and I get a warning when I run `pod install`.

**Comment by taviscaron** [comment]
Xcode requires that update starting from 8, not 8.1.
It happens because since Xcode 8 `EMBEDDED_CONTENT_CONTAINS_SWIFT` param is obsolete but still presented in .xcconfig files which are generated by cocoapods.
If you agree with Xcode and update settings - Xcode brakes cocoapods by setting `ALWAYS_EMBED_SWIFT_STANDARD_LIBRARIES` param which is replacement for `EMBEDDED_CONTENT_CONTAINS_SWIFT` but should be not set (or set to $(inherit)) since its value is brought from cocoapod's .xcconfig
Thus, I'm ignoring the warning about needed settings update and waiting cocoapods to eliminate obsolete `EMBEDDED_CONTENT_CONTAINS_SWIFT` from generated .xcconfig

**Comment by benasher44** [comment]
This should be fixed in CocoaPods 1.2.0.beta.1. It now only sets this flag to YES when required or doesn't set it at all. In the YES case, you can remove the setting from Xcode to resolve the conflict and remove the `pod install`/`pod update` warning. Can you try with the beta?

**Comment by taviscaron** [comment]
@benasher44 I confirm, there're no more warnings from Xcode with CocoaPods 1.2.0.beta.1.

**Comment by benasher44** [comment]
@davidbjames does 1.2.0.beta.1 fix the issue for you as well?

**Comment by migfabio** [comment]
@benasher44 I'm still having the warnings with 1.2.0.beta.1

**Comment by fitch** [comment]
I'm also getting this warning with 1.2.0.beta.1 just as @migfabio mentioned.

**Comment by benasher44** [comment]
@migfabio @fitch can you all be specific about which warnings you're seeing and on which projects or from what `pod` commands? There is a lot of "warning" discussing in this thread, which makes it hard to help. Thanks!

**Comment by taviscaron** [comment]
Xcode shows warning and recommends updating the project's settings. That actions breaks `pod install` that starts showing warnings like @davidbjames described above. If you revert these changes you will see Xcode warning about the needed update again and`pod install` will work well. 1.2.0.beta.1 fixes the reason Xcode requires update.

If you already have `pod install` warns then 1.2.0.beta.1 won't help you. You need to revert changes that were done by the Xcode or just follow inline recommendations.

**Comment by benasher44** [comment]
Once you make the changes that Xcode recommends, you shouldn't go revert all of the changes. That'll only cause Xcode to prompt you again. Once you've made the changes Xcode recommends, then go into Xcode and delete settings that `pod install` is complaining about. It's only warning you that you're overriding these settings (because following the Xcode prompt goes and sets them), and the end result will still be that the settings are set properly. It'll just be done by CocoaPods XCConfig files instead of Xcode project settings.

**Comment by Huang-Libo** [comment]
Yeah, problem solved after upgrade to **1.2.0.beta.1**

**Comment by dakeshi** [comment]
thx @benasher44 and community. I can solve my issue with your helpful comments.

In addition to @benasher44 comments, I think that it is helpful to give some description how to remove `ALWAYS_EMBED_SWIFT_STANDARD_LIBRARIES` in Xcode. 

1. Target-Build Setting
2. Choose `ALWAYS_EMBED_SWIFT_STANDARD_LIBRARIES`
3. press `Delete` button

If you follow these steps correctly, `ALWAYS_EMBED_SWIFT_STANDARD_LIBRARIES` setting key turns normal font size instead of bold size. 

Also, you can find out related answer in the [stackoverflow](http://stackoverflow.com/questions/39569743/errors-after-updating-to-xcode-8-no-such-module-and-target-overrides-the-em)

**Comment by benasher44** [comment]
Thanks folks! It sounds like this is fixed in 1.2.0, so I'm going to close.
